ml/Config.cc

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "Config.h"
+#include "ml-private.h"
+
+using namespace ml;
+
+/*
+ * Global configuration instance to be shared between training and
+ * prediction threads.
+ */
+Config ml::Cfg;
+
+template <typename T>
+static T clamp(const T& Value, const T& Min, const T& Max) {
+  return std::max(Min, std::min(Value, Max));
+}
+
+/*
+ * Initialize global configuration variable.
+ */
+void Config::readMLConfig(void) {
+    const char *ConfigSectionML = "ml";
+
+    bool EnableAnomalyDetection = config_get_boolean(ConfigSectionML, "enabled", false);
+    unsigned UpdateEvery = config_get_number(ConfigSectionML, "update every", 1);
+
+    /*
+     * Read values
+     */
+
+    int MaxTrainSecs = config_get_number(ConfigSectionML, "maximum num secs to train", 4 * 3600);
+    int MinTrainSecs = config_get_number(ConfigSectionML, "minimum num secs to train", 1 * 3600);
+    int TrainEvery = config_get_number(ConfigSectionML, "train every secs", 1 * 3600);
+
+    unsigned DiffN = config_get_number(ConfigSectionML, "num samples to diff", 1);
+    unsigned SmoothN = config_get_number(ConfigSectionML, "num samples to smooth", 3);
+    unsigned LagN = config_get_number(ConfigSectionML, "num samples to lag", 5);
+
+    unsigned MaxKMeansIters = config_get_number(ConfigSectionML, "maximum number of k-means iterations", 1000);
+
+    double DimensionAnomalyScoreThreshold = config_get_float(ConfigSectionML, "dimension anomaly score threshold", 0.99);
+    double HostAnomalyRateThreshold = config_get_float(ConfigSectionML, "host anomaly rate threshold", 0.01);
```

Comment:
@vkalintiris just double checking is `0.01` here 1%?

Reply:
It's ratio, ie. compared against `Num Anomalous Dimensions / Num total dimensions`.